### ISSUE TITLE
Disable spinner when using a debugger

### ISSUE THREAD

**Issue by blairfrandeen** [petereon/beaupy#93]
When using a debugging breakpoint when a spinner is running, the text entered in the debugger is continually obscured, while the spinner remains. This means that when debugging, I need to manually comment the spinner out of my code, or remember to type `spinner.stop()` when the debugger first comes up.

Minimum code to reproduce:
```python
import sys

from beaupy.spinners import DOTS, Spinner

spinner = Spinner(DOTS)
spinner.start()
breakpoint()
```

This could potentially be fixed by checking for
```python
sys.gettrace() is not None and sys.gettrace().__name__ == "trace_dispatch"
```
each time the spinner, cycles, and stopping the spinner if that's the case.

**Comment by petereon** [comment]
Thanks for a thorough issue submission and solution suggestion. I am not friendly with `pdb`, so it will take me some time to set-up and verify the solution. Considering I am presently a little pressed for time, I hope I can deliver a release with a fix towards the end of the week.

**Comment by petereon** [comment]
Yep, having refreshed my memory on the spinners code, it seems like a `rich` dependency issue. 

That said, I believe `get_renderable` callback could be extended with some sort of hack that would `stop` rendering of the Live display whenever we enter the debugging context, and `start` again once we leave it.

I'll see about setting up `pdb` and trying to fix this in `rich`. If that doesn't pan out, I'll try to hack it on top of `beaupy`

**Comment by petereon** [comment]
Sorry, got a little bogged down in life-stuff. This issue is not particularly easy to handle considering the layers and layers of abstraction that are required to deal with the terminal in some sort of elegant way that are included in `rich`. 

I've researched the option of hacking it on top of `beaupy` as well, but I couldn't figure out a somewhat acceptable way to do it.

With that in mind, I will probably not be able to get to this in a foreseeable future 😢 Nonetheless, I'll leave the issue open in case anyone else would like to have a crack at it in a meantime.

**Comment by tzah4748** [comment]
Wanted to drop my hacky solution to this issue.
I didn't test this too much, but it worked for me in the simple case.

```python
import sys
import threading
from dataclasses import dataclass, field
from threading import Thread
from time import sleep

from rich.status import Status

@dataclass
class DebuggableStatus:
    status: Status
    debug_detection_thread: Thread = field(init=False)

    def __post_init__(self):
        self.debug_detection_thread = Thread(target=self._detect_debug, daemon=True)
        self.debug_detection_thread.start()

    def _detect_debug(self):

        def is_debugger_running():
            for thread in threading.enumerate():
                if thread is threading.current_thread():
                    continue

                frame = sys._current_frames().get(thread.ident, None)
                while frame:
                    if frame.f_code.co_name == 'interaction':
                        return True
                    frame = frame.f_back

            return False

        while True:
            if is_debugger_running():
                self.status.stop()
            else:
                self.status.start()
            sleep(0.1)

    def __enter__(self) -> Status:
        return self.status.__enter__()

    def __exit__(self, exc_type, exc_val, exc_tb):
        return self.status.__exit__(exc_type, exc_val, exc_tb)
```

**Usage:**
```python
console = Console()
with DebuggableStatus(console.status('Running...')):
    breakpoint()
```
